src/psfmachine/perturbation.py

```diff
@@ -0,0 +1,364 @@
+            [self.time ** idx for idx in range(self.poly_order + 1)]
+        ).T
+        if self.cbvs:
+            self._get_cbvs()
+        if self.focus:
+            self._get_focus_change()
+        self._validate()
+        if self.segments:
+            self._cut_segments()
+        self._clean_vectors()
+
+    def __repr__(self):
+        return "PerturbationMatrix"
+
+    @property
+    def prior_mu(self):
+        return np.zeros(self.shape[1])
+
+    @property
+    def prior_sigma(self):
+        return np.ones(self.shape[1]) * 1e4
+
+    @property
+    def breaks(self):
+        return np.where(np.diff(self.time) / np.median(np.diff(self.time)) > 5)[0] + 1
+
+    def _validate(self):
+        # Check that the shape is correct etc
+        # Check the priors are right
+        if self.other_vectors is not None:
+            if isinstance(self.other_vectors, (list, np.ndarray)):
+                self.other_vectors = np.atleast_2d(self.other_vectors)
+                if self.other_vectors.shape[0] != len(self.time):
+                    if self.other_vectors.shape[1] == len(self.time):
+                        self.other_vectors = self.other_vectors.T
+                    else:
+                        raise ValueError("Must pass other vectors in the right shape")
+            else:
+                raise ValueError("Must pass a list as other vectors")
+            self.vectors = np.hstack([self.vectors, self.other_vectors])
+        return
+
+    def _cut_segments(self):
+        x = np.array_split(np.arange(len(self.time)), self.breaks)
+        masks = np.asarray(
+            [np.in1d(np.arange(len(self.time)), x1).astype(float) for x1 in x]
+        ).T
+        self.vectors = np.hstack(
+            [
+                self.vectors[:, idx][:, None] * masks
+                for idx in range(self.vectors.shape[1])
+            ]
+        )
+
+    def _get_focus_change(self, exptime=100):
+        focus = np.asarray(
+            [
+                np.exp(-exptime * (self.time - self.time[b]))
+                for b in np.hstack([0, self.breaks])
+            ]
+        )
+        focus *= np.asarray(
+            [
+                ((self.time - self.time[b]) >= 0).astype(float)
+                for b in np.hstack([0, self.breaks])
+            ]
+        )
+        self.vectors = np.hstack([self.vectors, np.nansum(focus, axis=0)[:, None]])
+        return
+
+    def _clean_vectors(self):
+        """Remove time polynomial from other vectors"""
+        nvec = self.poly_order + 1
+        if self.focus:
+            nvec += 1
+        if self.cbvs:
+            nvec += self.ncbvs
+        s = nvec * (len(self.breaks) + 1)
+        X = self.vectors[:, :s]
+        w = np.linalg.solve(X.T.dot(X), X.T.dot(self.vectors[:, s:]))
+        self.vectors[:, s:] -= X.dot(w)
+        return
+
+    def plot(self):
+        fig, ax = plt.subplots()
+        ax.plot(self.time, self.vectors + np.arange(self.shape[1]) * 0.1, c="k")
```

Comment:
this one crashes when calling from a `PerturbationMatrix3D` object because `self.shape` is different for `PerturbationMatrix3D` and `PerturbationMatrix`.

```suggestion
ax.plot(self.time, self.vectors + np.arange(self.vectors.shape[1]) * 0.1)
```

Reply:
colors will make easy to separate components.